src/azure-cli/azure/cli/command_modules/storage/operations/access_policy.py

```diff
@@ -7,13 +7,27 @@
 from azure.cli.core.profiles import ResourceType
 
 
+def _str_to_datetime(time):
+    form = '%Y-%m-%d'
+    try:
+        return datetime.strptime(time, form)
+    except ValueError:
+        pass
+    return time
+
+
 def create_acl_policy(cmd, client, policy_name, start=None, expiry=None, permission=None, **kwargs):
     """Create a stored access policy on the containing object"""
     t_access_policy = cmd.get_models('_models#AccessPolicy', resource_type=ResourceType.DATA_STORAGE_BLOB)
     acl = _get_acl(cmd, client, **kwargs)
+    if _get_service_container_type(cmd, client) == 'share':
+        signed_identifiers = {}
+        for identifier in acl["signed_identifiers"]:
+            signed_identifiers[identifier.id] = identifier.access_policy
+        acl = signed_identifiers
     acl[policy_name] = t_access_policy(permission if permission else '',
-                                       expiry if expiry else datetime.max,
-                                       start if start else datetime.utcnow())
+                                       _str_to_datetime(expiry) if expiry else datetime.max,
```

Comment:
This will also affect other services: queue, table, container. Arre you sure this works for all of them?